mysql/mysql.go

```diff
@@ -44,12 +45,37 @@ type URLOpener struct {
 
 // OpenMySQLURL opens a new database connection wrapped with OpenCensus instrumentation.
 func (uo *URLOpener) OpenMySQLURL(_ context.Context, u *url.URL) (*sql.DB, error) {
+	cfg, err := ConfigFromURL(u)
+	if err != nil {
+		return nil, err
+	}
 	return sql.OpenDB(connector{
-		dsn:       strings.TrimPrefix(u.String(), Scheme+"://"),
+		dsn:       cfg.FormatDSN(),
```

Comment:
Is this the bug fix? What's an example `dsn` value before the change and after it?

Reply:
In tests `mysql://root@localhost/mysql` was used but didn't work if you tried to send any sql command over the created db (received `default addr for network 'localhost' unknown` error).

The new value that is now used in tests is generated by terraform (e.g. `mysql://root:password@localhost:3306/testdb`) while the test ensures that this works in both `.Open` and `.Ping` calls.